hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/CompactionService.java

```diff
@@ -0,0 +1,184 @@
+        ozoneManager.getThreadNamePrefix());
+    this.ozoneManager = ozoneManager;
+    this.omMetadataManager = this.ozoneManager.getMetadataManager();
+
+    this.numCompactions = new AtomicLong(0);
+    this.suspended = new AtomicBoolean(false);
+    this.compactableTables = validateTables(tables);
+  }
+
+  private List<String> validateTables(List<String> tables) {
+    if (tables == null || tables.isEmpty()) {
+      return Collections.emptyList();
+    }
+    List<String> validTables = new ArrayList<>();
+    Set<String> allTableNames = new HashSet<>(omMetadataManager.listTableNames());
+    for (String table : tables) {
+      if (allTableNames.contains(table)) {
+        validTables.add(table);
+      } else {
+        LOG.warn("CompactionService: Table \"{}\" not found in OM metadata. Skipping this table.", table);
+      }
+    }
+    if (validTables.isEmpty()) {
+      LOG.error("CompactionService: No valid compaction tables found. Failing initialization.");
+      throw new IllegalArgumentException("CompactionService: None of the provided tables are valid.");
+    }
+    return Collections.unmodifiableList(validTables);
+  }
+
+  /**
+   * Suspend the service (for testing).
+   */
+  @VisibleForTesting
+  public void suspend() {
+    suspended.set(true);
+  }
+
+  /**
+   * Resume the service if suspended (for testing).
+   */
+  @VisibleForTesting
+  public void resume() {
+    suspended.set(false);
+  }
+
+  @VisibleForTesting
+  public List<String> getCompactableTables() {
+    return compactableTables;
+  }
+
+  /**
+   * Returns the number of manual compactions performed.
+   *
+   * @return long count.
+   */
+  @VisibleForTesting
+  public long getNumCompactions() {
+    return numCompactions.get();
+  }
+
+  @Override
+  public synchronized BackgroundTaskQueue getTasks() {
+    BackgroundTaskQueue queue = new BackgroundTaskQueue();
+    for (String tableName : compactableTables) {
+      queue.add(new CompactTask(tableName));
+    }
+    return queue;
+  }
+
+  private boolean shouldRun() {
+    return !suspended.get();
+  }
+
+  protected void compactFully(String tableName) throws IOException {
+    long startTime = Time.monotonicNow();
+    LOG.info("Compacting column family: {}", tableName);
+    try (ManagedCompactRangeOptions options = new ManagedCompactRangeOptions()) {
+      options.setBottommostLevelCompaction(ManagedCompactRangeOptions.BottommostLevelCompaction.kForce);
+      options.setExclusiveManualCompaction(true);
+      RocksDatabase rocksDatabase = ((RDBStore) omMetadataManager.getStore()).getDb();
+
+      try {
+        // Find CF Handler
+        RocksDatabase.ColumnFamily columnFamily = rocksDatabase.getColumnFamily(tableName);
+        rocksDatabase.compactRange(columnFamily, null, null, options);
+        LOG.info("Compaction of column family: {} completed in {} ms",
+            tableName, Time.monotonicNow() - startTime);
+      } catch (NullPointerException ex) {
```

Comment:
We should check null instead of catching `NullPointerException`.

BTW, the commit message somehow used HDDS-12518 instead of [HDDS-12819](https://github.com/apache/ozone/pull/8260/files#top).
```
commit b91e8e732f07e220b1be6b64fb74a8d3f67dbf3c
Author: Tejaskriya <87555809+Tejaskriya@users.noreply.github.com>
Date:   Wed Apr 16 13:19:49 2025 +0530

    HDDS-12518. Auto-compact tables which can tend to be large in size at intervals (#8260)
```